src/graph/sampling/randomwalks_cpu.h

```diff
@@ -0,0 +1,75 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/sampler/generic_randomwalk_cpu.h
+ * \brief DGL sampler - templated implementation definition of random walks on CPU
+ */
+
+#ifndef DGL_GRAPH_SAMPLING_RANDOMWALKS_CPU_H_
+#define DGL_GRAPH_SAMPLING_RANDOMWALKS_CPU_H_
+
+#include <dgl/base_heterograph.h>
+#include <dgl/array.h>
+#include "randomwalks.h"
+
+namespace dgl {
+
+using namespace dgl::runtime;
+using namespace dgl::aten;
+
+namespace sampling {
+
+namespace impl {
+
+namespace {
+
+/*!
+ * \brief Generic Random Walk.
+ * \param hg The heterograph.
+ * \param seeds A 1D array of seed nodes, with the type the source type of the first
+ *        edge type in the metapath.
+ * \param max_num_steps The maximum number of steps of a random walk path.
+ * \param step The random walk step function with type \c StepFunc.
+ * \return A 2D array of shape (len(seeds), max_num_steps + 1) with node IDs.
+ */
+template<DLDeviceType XPU, typename IdxType>
+IdArray GenericRandomWalk(
+    const HeteroGraphPtr hg,
```

Comment:
`hg` is not used in this function. I guess it is bound in the closure of `StepFunc`. Should we remove this arg?